app/src/main/java/com/example/sharewhatyoucanproject/deliverfood/DeliverViewModel.kt

```diff
@@ -0,0 +1,117 @@
+package com.example.sharewhatyoucanproject.deliverfood
+
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import com.example.sharewhatyoucanproject.models.GeoPoint
+import com.example.sharewhatyoucanproject.models.PostModel
+import com.example.sharewhatyoucanproject.models.RequestModel
+import com.google.firebase.auth.FirebaseAuth
+import com.google.firebase.firestore.FieldValue
+import com.google.firebase.firestore.FirebaseFirestore
+import java.util.*
+
+class DeliverViewModel(
+    private val db: FirebaseFirestore,
+) : ViewModel() {
+
+    private val _deliverResult = MutableLiveData<DeliverResult>()
+    val deliverResult: MutableLiveData<DeliverResult> = _deliverResult
+
+    private val _donationResult = MutableLiveData<DonationResult>()
+    val donationResult: MutableLiveData<DonationResult> = _donationResult
+
+    fun getItems() {
+        db.collection("posts")
+            .whereEqualTo("uid", FirebaseAuth.getInstance().currentUser!!.uid)
+            .whereNotEqualTo("status", 2)
+            .get()
+            .addOnCompleteListener { task ->
+                if (task.isSuccessful) {
+                    val list = mutableListOf<PostModel>()
+                    for (ds in task.result) {
+                        val postModel = ds.getGeoPoint("location")?.let {
+                            PostModel(
+                                ds.getString("imageUrl")!!,
```

Comment:
You can use toString here to avoid double bangs and adding empty strings

Reply:
done